Frontend/src/app/core/data.service.ts

```diff
@@ -29,8 +29,8 @@ export class DataService {
         const cache = this.getCache<T>(path);
         const url = this.getUrl(path);
 
-        this.authHttpService.get(url)
-            .map(response => response.json() as T[])
+        this.authHttpService.get(url).pipe(
+            map(response => response as unknown as T[]))
```

Comment:
Why is this needed? (repeated across file)

Reply:
- In HttpClient, JSON is an assumed default and no longer needs to be explicitly parsed
- According to response, it is an Array of Objects but missing some properties from the normal array in javascript, like pop, push and concat. so I force typecast it discarding the remaining properties

Reply:
I meant use get <T[]>() instead